server/channels/api4/user.go

```diff
@@ -1965,6 +1966,31 @@ func login(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func loginWithDesktopToken(c *Context, w http.ResponseWriter, r *http.Request) {
+	props := model.MapFromJSON(r.Body)
+	clientToken := props["client_token"]
+	serverToken := props["server_token"]
+	deviceId := props["device_id"]
+
+	user, err := c.App.ValidateDesktopToken(clientToken, serverToken, time.Now().Add(-model.DesktopTokenTTL).Unix())
```

Comment:
It adds that time negative to the time requested. So if it's 3:03PM it becomes 3:00PM
We do this to check if the created at date is after this time so that we know if the token is expired or not.

Reply:
I wonder if this is responsibility of the caller or responsibility of the app layer, but 0/5